instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -177,22 +178,11 @@ def process_view(self, request, view_func, *args, **kwargs):
                         span.set_attribute("http.route", route)
 
     def process_exception(self, request, exception):
-        # Django can call this method and process_response later. In order
-        # to avoid __exit__ and detach from being called twice then, the
-        # respective keys are being removed here.
         if self._excluded_urls.url_disabled(request.build_absolute_uri("?")):
             return
 
         if self._environ_activation_key in request.META.keys():
-            request.META[self._environ_activation_key].__exit__(
```

Comment:
Wasn't this behavior already implemented here? How does moving the logic to `process_response` change anything?

Reply:
It was but because we popped the activation key, process_response did not do anything. We recorded the exception but ignored the response meaning response code or metrics were not recorded for example. We could either duplicate the code here, take it out in a function and call from both places or just make the exception available to process_response so it can handle all cases.

Reply:
Right makes sense. Doesn't `add_response_attributes` already handle setting the status and `http.status_code` though?

Reply:
Right, it does but without this change it was never executed in case of an exception so status_code (and other response attrs) were never set in case of an exception.